app/models/team.rb

```diff
@@ -247,18 +251,11 @@ def self.export(csv, parent_id, options, teamtype)
   end
 
   # Create the team with corresponding tree node
-  def self.create_team_and_node(id, teamtype = 'AssignmentTeam')
-    if teamtype == 'CourseTeam'
-      curr_course = Course.find(id)
-      team_name = Team.generate_team_name(curr_course.name)
-      team = CourseTeam.create(name: team_name, parent_id: id)
-      TeamNode.create(parent_id: id, node_object_id: team.id)
-    elsif teamtype == 'AssignmentTeam'
-      curr_assignment = Assignment.find(id)
-      team_name = Team.generate_team_name(curr_assignment.name)
-      team = AssignmentTeam.create(name: team_name, parent_id: id)
-      TeamNode.create(parent_id: id, node_object_id: team.id)
-    end
+  def self.create_team_and_node(id)
+    current_task = parent_model_for_id id
+    team_name = Team.generate_team_name(current_task.name)
+    team = self.create(name: team_name, parent_id: id)
+    TeamNode.create(parent_id: id, node_object_id: team.id)
```

Comment:
The code here is tricky...
The `parent_id` will be the id of the 'parent_model' and the node_object_id will be the id of newly created team. 
Comment should be added to help future developers understand this part.